### DISCUSSION TITLE
Linter and formater

### DISCUSSION THREAD

**Discussion by mikededo** [github-stats-22/github-stats#2] [Ideas]
## Definition of a linter and a formatter

Not really aware of how it works on the Java world, but for the little I have searched, there are plently of code linters and code formatters for Java code.

From here, I would like to discuss on which linter and formatter to use so that we can ensure that our code follows "some rules", it is structured the same way and it is beatuifully formatted 💅🏼!

How about...?

- [Google Java Format](https://plugins.jetbrains.com/plugin/8527-google-java-format)

**Comment by plozanol** [comment]
Other interesting libraries and references:

https://maven.apache.org/plugins/maven-checkstyle-plugin/index.html
https://github.com/google/google-java-format
https://google.github.io/styleguide/javaguide.html

**Comment by sdomingobasora** [comment]
@cjp4 has suggested using [checkstyle](https://checkstyle.sourceforge.io/), configurable via XML, looks good to me.

**Comment by plozanol** [comment]
I was reviewing the diff of checkstyle.xml that @cjp4 uploaded at the "configuration base" commit against the official latest:
 https://github.com/mikededo/github-stats/blob/20ee0ca5812e4dd31ee69e9b017ad375090b15e0/checkstyle.xml

and seems a bit outdated. I propose to start with the official one and let's tweak the config with our flavor to this project.

https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml

**Comment by mikededo** [comment]
What about sharing an IntelliJ formatting configuration? We all use IntelliJ so we could either use an official guide (such as the Google style guide) or somemone could share their configuration and the others just set it up.
